Plumbing Reagent UI changes & fixes[Take 3]

SyncIt21 opened the pull request:
**About the pull request**
3rd attempt cause the 1st [PR](https://github.com/tgstation/tgstation/pull/71032) had way too many problems and the 2nd [PR](https://github.com/tgstation/tgstation/pull/71101) had merge conflicts and became stale so fuck it delete those branches & try again.

1)For Plumbing the 
- Chemical Reaction Chamber
- Chemical Filter
- As a side effect for Modifying Chemical Filter the BloodFilter also now

Picks reagents from an input-list with a search bar attached at the bottom 

2)Fixes bug which prevented the user from entering new values[temperature,acidic ph & alkaline ph] into the ChemReactionChamber. They would be stuck at their defaults 300k,5,9 respectively even if you tried to enter new values it would reject those new values. That's fixed now


https://user-images.githubusercontent.com/110812394/203076880-eb801c76-68ec-4470-8648-9239270fbcbc.mp4

**Why its good for the game**
1. Removes the need to type long reagent names manually by hand which is error prone and time consuming
2. Allows you to change the ph  & temp values for various reactions which would have not been possible if not for this

**ChangeLog**
:cl:

qol: Entering reagent names is much faster with zero errors

fix: You can enter new temp & ph values

/:cl:

SyncIt21 committed:
UI & Fix

1)Reaction Chamber ,ChemFilter ,BloodFilter all pick from input list

2)Fixed a bug which prevented entering new values for temperature , acidic ph & basic ph in reaction chamber

SyncIt21 commented:
@Time-Green sorry to trouble you the 3rd time with this but it's finalized now. This was my very first PR and my 1st time using github so naturally i messed up. No more

SyncIt21 commented:
> > @Time-Green sorry to trouble you the 3rd time with this but it's finalized now. This was my very first PR and my 1st time using github so naturally i messed up. No more
> 
> I've been using github for 8 years and I still fuck up frequently, I'm just glad you succeeded eventually

:D

Time-Green commented on code/modules/plumbing/plumbers/filter.dm:
```diff
@@ -37,20 +37,24 @@
-							right += chem_id
-			else
-				to_chat(usr, span_warning("No such known reagent exists!"))
+			var/which=params["which"]
```
```suggestion
			var/which = params["which"]
```

Time-Green commented on code/modules/plumbing/plumbers/reaction_chamber.dm:
```diff
@@ -84,27 +85,37 @@
-		return
+		return TRUE
+
+	.=FALSE
```
```suggestion
	. = FALSE
```

Time-Green commented on code/modules/plumbing/plumbers/reaction_chamber.dm:
```diff
@@ -84,27 +85,37 @@
 					required_reagents[input_reagent] = input_amount
-					. = TRUE
+
+			.=TRUE
```
```suggestion
			. = TRUE
```

Time-Green commented on code/modules/plumbing/plumbers/reaction_chamber.dm:
```diff
@@ -84,27 +85,37 @@
+			var/reagent = get_chem_id(params["chem"])
+			if(reagent)
+				required_reagents.Remove(reagent)
+			.=TRUE
```
```suggestion
			. = TRUE
```

Time-Green commented on code/modules/plumbing/plumbers/reaction_chamber.dm:
```diff
@@ -84,27 +85,37 @@
+			var/target = text2num(params["target"])
+			if(target != null)
+				target_temperature=clamp(target,0,1000)
+			.=TRUE
```
```suggestion
			. = TRUE
```

Time-Green commented on code/modules/plumbing/plumbers/reaction_chamber.dm:
```diff
@@ -84,27 +85,37 @@
-				target_temperature = clamp(target, 0, 1000)
+			var/target = text2num(params["target"])
+			if(target != null)
+				target_temperature=clamp(target,0,1000)
```
```suggestion
				target_temperature=clamp(target, 0, 1000)
```

Time-Green commented on code/modules/plumbing/plumbers/reaction_chamber.dm:
```diff
@@ -154,17 +165,17 @@
 	.["ph"] = round(reagents.ph, 0.01)
 	.["reagentAcidic"] = acidic_limit
 	.["reagentAlkaline"] = alkaline_limit
-	return .
```
im dying who even added this

SyncIt21 replied:
OOOFFF

Time-Green commented on code/modules/reagents/chemistry/holder.dm:
```diff
@@ -20,6 +20,15 @@
+/proc/init_chemical_name_list()
+	var/list/name_list = list()
+	for(var/X in GLOB.chemical_reagents_list)
+		var/datum/reagent/R = GLOB.chemical_reagents_list[X]
```
Try not to use one letter variable names

SyncIt21 replied:
Got it

Time-Green reviewed (changes requested):
Nice work! Try to use spaces in expressions and stuff though, it helps with readability

SyncIt21 committed:
Update code/modules/plumbing/plumbers/filter.dm

Co-authored-by: Time-Green <timkoster1@hotmail.com>

SyncIt21 committed:
Update code/modules/plumbing/plumbers/reaction_chamber.dm

Co-authored-by: Time-Green <timkoster1@hotmail.com>

SyncIt21 committed:
Update code/modules/plumbing/plumbers/reaction_chamber.dm

Co-authored-by: Time-Green <timkoster1@hotmail.com>

SyncIt21 committed:
Update code/modules/plumbing/plumbers/reaction_chamber.dm

Co-authored-by: Time-Green <timkoster1@hotmail.com>

SyncIt21 committed:
Update code/modules/plumbing/plumbers/reaction_chamber.dm

Co-authored-by: Time-Green <timkoster1@hotmail.com>

SyncIt21 committed:
Update holder.dm

Renamed variable R to Reagent

SyncIt21 commented:
Merged your commit 4 times into reaction chamber.dm cause I forgot to refresh the page. My bad

SyncIt21 committed:
Merge branch 'master' into Plumbing-UI

Time-Green commented:
> Merged your commit 4 times into reaction chamber.dm cause I forgot to refresh the page. My bad

That happens. You can also just manually commit those suggestions yourself, it’s not like I suggested anything groundbreaking

SyncIt21 commented:
Thanks I'm all done here. I think, hope

SyncIt21 committed:
Merge branch 'master' into Plumbing-UI

stylemistake reviewed (commented):
Suggestions to reduce text and make UI a bit more compact.

stylemistake commented on tgui/packages/tgui/interfaces/ChemFilter.tsx:
```diff
@@ -11,37 +11,30 @@ type Data = {
   list: string[];
-  reagentName: string;
-  onReagentInput: (str: string) => void;
+  buttonTitle: string;
```
I don't think custom title is necessary at all, since it will always be adding reagents, regardless of the context.

```suggestion
```

stylemistake commented on tgui/packages/tgui/interfaces/ChemFilter.tsx:
```diff
@@ -11,37 +11,30 @@ type Data = {
 export const ChemFilterPane = (props: Props, context) => {
   const { act } = useBackend(context);
-  const { title, list, reagentName, onReagentInput } = props;
+  const { title, list, buttonTitle, buttonColor } = props;
```
```suggestion
  const { title, list, buttonColor } = props;
```

SyncIt21 commented on tgui/packages/tgui/interfaces/BloodFilter.tsx:
```diff
@@ -20,8 +19,8 @@ export const BloodFilter = (props, context) => {
               list={whitelist}
-              reagentName={chemName}
-              onReagentInput={(value) => setChemName(value)}
+              buttonTitle="Blood White List"
```
I can't remove buttonTitle as this variable is used by ChemFilter.tsx to display text for this button. Maybe i just do `buttonTitle="Reagent Filter List"` as the name implies reagents to filter from blood. I thought since it's about blood give it a red color. I'll give it green color instead

SyncIt21 replied:
Sure i'll delete the variable buttonTitle.  I thought red color is assosiated with blood but maybe  green color is nice

SyncIt21 committed:
@stylemistake : changes made

1. removed variable buttonTitle and replaced it with just "Add reagent"
2. replaced width="150px" with icon="plus"

SyncIt21 commented:
@stylemistake all changes made. Anything else?

SyncIt21 committed:
Merge branch 'master' into Plumbing-UI

RikuTheKiller commented:
may we have a bar for typing in names as WELL as the dropdown menu? i personally liked typing them out since its faster than having to click a button and then typing it out

SyncIt21 commented:
There is a serach bar attached at the bottom of the drop down list For you to type ,Besides attaching an bar next to the button would look ugly and no one would approve that PR. The main purpose of this PR was to reduce typing errors and most ppl will prefer that over typing speed so I'm afraid the majority wins
